code/modules/power/cable.dm

```diff
@@ -549,32 +586,173 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list(new/datum/stack_recipe("cable restrain
 	custom_materials = list()
 	cost = 1
 
-/obj/structure/cable_bridge
-	name = "cable bridge"
-	desc = "A bridge to connect different cable layers, or link terminals to incompatible cable layers."
+#undef UNDER_SMES
+#undef UNDER_TERMINAL
+
+///multilayer cable to connect different layers
+/obj/structure/cable/multilayer
+	name = "multilayer cable hub"
+	desc = "A flexible, superconducting insulated multilayer hub for heavy-duty multilayer power transfer. Refuse direct connections to other hubs."
 	icon = 'icons/obj/power.dmi'
 	icon_state = "cable_bridge"
-	layer = WIRE_LAYER + 0.02 //Above all the cables but below terminals
-	anchored = TRUE
-	obj_flags = CAN_BE_HIT | ON_BLUEPRINTS
+	cable_layer = CABLE_LAYER_2
+	machinery_layer = MACHINERY_LAYER_1
+	layer = WIRE_LAYER - 0.02 //Below all cables Disabled layers can lay over hub
+	color = "white"
+	var/obj/effect/node/machinery_node
+	var/obj/effect/node/layer1/cable_node_1
+	var/obj/effect/node/layer2/cable_node_2
+	var/obj/effect/node/layer3/cable_node_3
+
+/obj/effect/node
+	icon = 'icons/obj/power_cond/layer_cable.dmi'
+	icon_state = "l2-noconnection"
+	vis_flags = VIS_INHERIT_ID|VIS_INHERIT_PLANE|VIS_INHERIT_LAYER
+	color = "black"
 
-/obj/structure/cable_bridge/Initialize(mapload)
-	. = ..()
-	if (!mapload)
-		var/first = TRUE
-		var/datum/powernet/PN
-		for(var/obj/structure/cable/C in get_turf(src))
-			C.update_icon()
-			if(first == TRUE)
-				first = FALSE
-				PN = C.powernet
-				continue
-			propagate_network(C, PN)
+/obj/effect/node/layer1
+	color = "red"
+	icon_state = "l1-1-2-4-8-node"
+	vis_flags = VIS_INHERIT_ID|VIS_INHERIT_PLANE|VIS_INHERIT_LAYER|VIS_UNDERLAY
+
+/obj/effect/node/layer2
+	color = "yellow"
+	icon_state = "l2-1-2-4-8-node"
+	vis_flags = VIS_INHERIT_ID|VIS_INHERIT_PLANE|VIS_INHERIT_LAYER|VIS_UNDERLAY
+
+/obj/effect/node/layer3
+	color = "blue"
+	icon_state = "l4-1-2-4-8-node"
+	vis_flags = VIS_INHERIT_ID|VIS_INHERIT_PLANE|VIS_INHERIT_LAYER|VIS_UNDERLAY
+
+/obj/structure/cable/multilayer/update_icon_state()
+	return
 
-/obj/structure/cable_bridge/wirecutter_act(mob/living/user, obj/item/I)
+/obj/structure/cable/multilayer/update_icon()
+	var/R = 50
+	var/G = 50
+	var/B = 50
+
+	R += cable_layer & CABLE_LAYER_1 ? 150:0
+	G += cable_layer & CABLE_LAYER_2 ? 150:0
+	B += cable_layer & CABLE_LAYER_3 ? 150:0
+
+	color = rgb(R, G, B)
+
+	machinery_node?.alpha = machinery_layer & MACHINERY_LAYER_1 ? 255 : 0

```

Comment:
```suggestion
	if(machinery_node)
		machinery_node.alpha = machinery_layer & MACHINERY_LAYER_1 ? 255 : 0
```
Else you can end up with a `null = something` operation.

Reply:
Online DM ref cant open ?. page so
![image](https://user-images.githubusercontent.com/7734424/78066277-4a30e700-739d-11ea-8ed0-0ca6c15636a3.png)

Reply:
Bit ugly, but you are right.